Support vllm quantization

ivanvykopal opened the pull request:
## Title

I have implemented a feature for loading quantized models with vellum. The current version does not support quantized models with vllm.

## Relevant issues

There is no relevant issue.

## Type

🆕 New Feature

## Changes

I changed the `validate_environment` in `vllm/completion/handler.py` to support the loading of quantized versions of models. It was done by providing several vllm's default parameters and updating them with parameters from `optional_params`.

## [REQUIRED] Testing - Attach a screenshot of any new tests passing locall
If UI changes, send a screenshot/GIF of working UI fixes

<!-- Test procedure -->

ivanvykopal committed:
feat: support vllm quantization

krrishdholakia commented on litellm/llms/vllm/completion/handler.py:
```diff
@@ -27,14 +27,31 @@ def __init__(self, status_code, message):
 
 # check if vllm is installed
-def validate_environment(model: str):
+def validate_environment(model: str, optional_params: Union[dict, None]):
```
this should be implemented in the transformation.py - https://github.com/BerriAI/litellm/blob/246e3bafc89e19297fa42aabbfca6d25058e5989/litellm/llms/vllm/completion/transformation.py#L14

krrishdholakia commented on litellm/llms/vllm/completion/handler.py:
```diff
@@ -142,7 +159,7 @@ def batch_completions(
     """
     try:
-        llm, SamplingParams = validate_environment(model=model)
+        llm, SamplingParams, optional_params = validate_environment(model=model, optional_params=optional_params)
```
can you add a mock test w/ screenshot of this working?

similar to this - https://github.com/BerriAI/litellm/blob/246e3bafc89e19297fa42aabbfca6d25058e5989/tests/llm_translation/test_azure_ai.py#L48


ideally this test would **not** add the vllm sdk as a dep on the ci/cd pipeline (so maybe use a magicmock object here)

ivanvykopal committed:
feat: move default params to VLLMConfig

ivanvykopal committed:
feat: add tests for vllm

ivanvykopal commented:
Hi @krrishdholakia, thank you for your comments.

I have moved the default parameters to `transformation.py` and also added tests for vllm, similar to Azure, as you provided an example.

Here is the screenshot of the tests.
<img width="1041" alt="image" src="https://github.com/user-attachments/assets/bb97c42b-28fb-4050-85ca-d1ae7114514d" />

ivanvykopal committed:
Merge branch 'main' into vllm-quantization

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=7297) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=7297) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=7297) it.</sub>
